app/javascript/components/patient/PatientReportsTable.js

```diff
@@ -0,0 +1,372 @@
+          { label: 'Actions', field: '', isSortable: false, filter: this.createActionsButton },
+          { label: 'ID', field: 'id', isSortable: true },
+          { label: 'Needs Review', field: 'symptomatic', isSortable: true },
+          { label: 'Reporter', field: 'who_reported', isSortable: true },
+          { label: 'Created At', field: 'created_at', isSortable: true, filter: this.formatTimestamp },
+        ],
+        rowData: [],
+        totalRows: 0,
+        selectedRows: [],
+        selectAll: false,
+      },
+      query: {
+        page: 0,
+        entries: 10,
+      },
+      entryOptions: [10, 15, 25],
+      cancelToken: axios.CancelToken.source(),
+      isLoading: false,
+      editRow: null,
+      actionsEnabled: false,
+      showReportModal: false,
+    };
+  }
+
+  componentDidMount() {
+    this.populateSymptomCols();
+    this.updateTable(this.state.query);
+  }
+
+  /**
+   * Handles change of input in the search bar. Updates table based on input.
+   * @param {SyntheticEvent} event - Event when the search input changes
+   */
+  handleSearchChange = event => {
+    const value = event.target.value;
+    this.setState(
+      state => {
+        return { query: { ...state.query, search: value } };
+      },
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  populateSymptomCols = () => {
+    for (const symptom of this.props.symptoms) {
+      const symptom_col = { label: symptom.label, field: symptom.name, isSortable: true, filter: this.filterSymptomCell };
+      this.setState(state => {
+        const updated_table_data = { ...state.table };
+        updated_table_data.colData.push(symptom_col);
+        return {
+          table: updated_table_data,
+        };
+      });
+    }
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  filterSymptomCell = (value, rowData, colData, rowIndex, colIndex) => {
+    const symptomName = colData.field;
+    const passesThreshold = rowData.passes_threshold_data[symptomName.toString()];
+    const className = passesThreshold ? 'concern' : '';
+    return <span className={className}>{value}</span>;
+  };
+
+  /**
+   * Called when table data is to be updated because of some change to the table setting.
+   * @param {Object} query - Updated query for table data after change.
+   */
+  updateTable = query => {
+    // cancel any previous unfinished requests to prevent race condition inconsistencies
+    this.state.cancelToken.cancel();
+
+    // generate new cancel token for this request
+    const cancelToken = axios.CancelToken.source();
+
+    this.setState({ query, cancelToken, isLoading: true }, () => {
+      this.queryServer(query);
+    });
+  };
+
+  /**
+   * Returns updated table data via an axios POST request.
+   */
+  queryServer = _.debounce(query => {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios
+      .post('/patients/' + this.props.patient.id + '/assessments/table', {
```

Comment:
Does this one actually need to be a POST?

Reply:
I made it a POST To be consistent with the line list request, but given it's not really necessary and we want to use `index` for the route endpoint, I've updated it to be a GET here: https://github.com/SaraAlert/SaraAlert/pull/604/commits/30d4fbeb075c861ae8a66567d663f31a9b23ee50